pkg/controller/scylladbmonitoring/sync_grafana_test.go

```diff
@@ -125,3 +130,97 @@ spec:
 		})
 	}
 }
+
+func Test_makeGrafanaDashboards(t *testing.T) {
+	tt := []struct {
+		name           string
+		sm             *scyllav1alpha1.ScyllaDBMonitoring
+		expectedString string
+		expectedErr    error
+	}{
+		{
+			name: "renders data for default SaaS type",
+			sm: &scyllav1alpha1.ScyllaDBMonitoring{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "sm-name",
+				},
+				Spec: scyllav1alpha1.ScyllaDBMonitoringSpec{
+					Type: nil,
+				},
+			},
+			expectedString: strings.TrimLeft(`
+apiVersion: v1
+data:
+  overview.json.gz.base64: <13216 bytes>
+kind: ConfigMap
+metadata:
+  creationTimestamp: null
+  name: sm-name-grafana-scylladb-dashboards
+`, "\n"),
+			expectedErr: nil,
+		},
+		{
+			name: "renders data for platform type",
+			sm: &scyllav1alpha1.ScyllaDBMonitoring{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "sm-name",
+				},
+				Spec: scyllav1alpha1.ScyllaDBMonitoringSpec{
+					Type: pointer.Ptr(scyllav1alpha1.ScyllaDBMonitoringTypePlatform),
+				},
+			},
+			expectedString: strings.TrimLeft(`
+apiVersion: v1
+data:
+  advanced.json.gz.base64: <10636 bytes>
```

Comment:
the primary intend is that you write about the expected amount of data, say it had a bug when not closing the gzip writer and always writing just 16b. Uncompressed content might be better but it's not practically possible.

Reply:
comparing just length of compressed data is error prone as it doesn't detect content changes if compressed data takes the same amount of space. Hash of gziped content would be better.

Reply:
I've used hashes initially and when I forgot to close the gzip writer I didn't noticed it was just 16 bytes so the count actually carries more information then a hash.

Reply:
And the test with hash comparison succeeded with this bug? The only situation where such test would pass is when dev takes the golden hash from the test output. Golden hash shouldn't be taken from implementation we test, but from reference implementation, in this case gzip + base64 utilities. 
I'm pointing out that length comparison has a flaw. Hash doesn't have it.

Reply:
also the gzip byte count is what matters for whether it fits into a CM, so it can be helpful to know how much the bump is raising it. I hope we don't get close though.

Reply:
I'm not suggesting to remove it, hash and length check can live together.

Reply:
> And the test with hash comparison succeeded with this bug?

well, the hash noticed the change but it was expected as the file was moved from plain text to gzip - the change is all the info you get from a hash and change was expected (not the data size)

Reply:
added extra hash